src/ophyd_async/core/device.py

```diff
@@ -32,6 +33,8 @@ class Device(HasName):
     _name: str = ""
     #: The parent Device if it exists
     parent: Optional[Device] = None
+    # None if connect hasn't started, an Event if it has, a set Event if it's done
+    _connect_task: Optional[asyncio.Task] = None
```

Comment:
```suggestion
    # Previous connect was successful, False on initialisation since there was
    # no previous connect
    _previous_connect_ok: bool = False
```